### PR TITLE
Adds custom policy type, fixes invalid policy error

### PR DESCRIPTION
**Pull request by cameronvoell**
Related to https://github.com/xmtp/libxmtp/pull/867

### CODE REVIEW

File: library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt [hunk 1]
```diff
@@ -485,7 +484,7 @@ class GroupTest {
 
         Thread.sleep(1000)
 
-        assertEquals(4, messageCallbacks)
+        assertEquals(secondMsgCheck, messageCallbacks)
```
Comment (cameronvoell) [inline library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt:487]: this test was failing before the change, think this last value is supposed to also be secondMsgCheck

File: library/src/main/java/xmtpv3.kt [hunk 2]
```diff
@@ -47,17 +47,14 @@ import kotlinx.coroutines.suspendCancellableCoroutine
 open class RustBuffer : Structure() {
     // Note: `capacity` and `len` are actually `ULong` values, but JVM only supports signed values.
     // When dealing with these fields, make sure to call `toULong()`.
-    @JvmField
-    var capacity: Long = 0
-    @JvmField
-    var len: Long = 0
-    @JvmField
-    var data: Pointer? = null
+    @JvmField var capacity: Long = 0
```
Comment (nplasterer) [inline library/src/main/java/xmtpv3.kt:50]: Can you run `command + option + L` on this file to format it. That should bring down the changes.

**Review by nplasterer — APPROVED** [review]
Don't forget to copy over this file too https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/libxmtp-version.txt

File: library/src/main/jniLibs/armeabi-v7a/libuniffi_xmtpv3.so
Comment (nplasterer) [inline library/src/main/jniLibs/armeabi-v7a/libuniffi_xmtpv3.so:1]: I feel mildly concerned that there is only removals on most of these files.

Comment (cameronvoell) [reply library/src/main/jniLibs/armeabi-v7a/libuniffi_xmtpv3.so:1]: re-ran crossbuild locally, and had no errors, and files were the same. also tested locally running tests and against RN. Agree getting these built on CI would be a big improvement though 👍

File: library/src/main/java/xmtpv3.kt (hunk 2 above)
Comment (cameronvoell) [reply library/src/main/java/xmtpv3.kt]: version file added here 247d07cf8483792097f2017ea9e7f7431ba1e706

**Pull request merged**